lib/wasix/src/runtime/resolver/wapm_source.rs

```diff
@@ -86,9 +86,9 @@ impl Source for WapmSource {
 
         for pkg_version in versions {
             tracing::trace!(?pkg_version, "checking package version");
-
             let version = Version::parse(&pkg_version.version)?;
-            if version_constraint.matches(&version) {
+
+            if !pkg_version.is_archived && version_constraint.matches(&version) {
```

Comment:
This is suboptimal for UX. 

Can we: 

* if the constraint matches the version exactly (`=1.1.1`), show an error that the given version was archived
* if the the constraint matches loosely (eg `1.1.1`), and there is no other available version compatible version, show an error that the only matching version was archived?

* produce a terminal debug message that an archived version was skipped, for debugging?

Reply:
Yeah, printing a debug message when we skip over an archived package version is a good idea :+1:

I'm quite hesitant to start poking around inside `semver::VersionReq` and adding special cases, though. While it might feel convenient, that also opens us up to some pretty magical behaviour.

My thinking is that it's easy to give people a single rule that lockfiles reuse previous dependency resolution results and archiving a package will always remove it from being considered when doing future dependency resolution. 

However, things get a lot muddier once you start saying archived versions are almost always ignored, but we'll also include it if you add a strict `=` constraint, or if you didn't precisely ask for that version but there happened to be no other versions in that range.

Reply:
We had a chat about this during today's standup and @syrusakbary feels the same way about adding edge cases.

The redirect for `Accept: application/webc` will always work for archived packages, so if you want you can always do `wasmer run https://wasmer.io/python/python@1.2.3` if you want.

Reply:
> if the constraint matches the version exactly (=1.1.1), show an error that the given version was archived

What I wrote above is not that these situations should allow usage of the archived version, I was just referring to producing concrete errors, which state  specific version is archived (`=xxx`) and thus can't be used, or that the only matching version is archived, rather than just saying "could not find a matching version"

Reply:
Ah, gotcha. I've added something that should error out under those conditions.